pkg/reconciler/events/scheduler/scheduler.go

```diff
@@ -63,117 +58,29 @@ type Reconciler struct {
-		logging.FromContext(ctx).Desugar().Warn("Failed to update CloudSchedulerSource status", zap.Error(uErr))
-		r.Recorder.Eventf(scheduler, corev1.EventTypeWarning, "UpdateFailed",
-			"Failed to update status for CloudSchedulerSource %q: %v", scheduler.Name, uErr)
-		return uErr
-	} else if reconcileErr == nil {
-		// There was a difference and updateStatus did not return an error.
-		r.Recorder.Eventf(scheduler, corev1.EventTypeNormal, "Updated", "Updated CloudSchedulerSource %q", scheduler.Name)
-	}
-	if reconcileErr != nil {
-		r.Recorder.Event(scheduler, corev1.EventTypeWarning, "InternalError", reconcileErr.Error())
-	}
-	return reconcileErr
-}
+// Check that our Reconciler implements Interface.
+var _ cloudschedulersourcereconciler.Interface = (*Reconciler)(nil)
 
-func (r *Reconciler) reconcile(ctx context.Context, scheduler *v1alpha1.CloudSchedulerSource) error {
+func (r *Reconciler) ReconcileKind(ctx context.Context, scheduler *v1alpha1.CloudSchedulerSource) reconciler.Event {
 	ctx = logging.WithLogger(ctx, r.Logger.With(zap.Any("scheduler", scheduler)))
 
 	scheduler.Status.InitializeConditions()
-
-	// See if the source has been deleted.
-	if scheduler.DeletionTimestamp != nil {
-		logging.FromContext(ctx).Desugar().Debug("Deleting CloudSchedulerSource job")
-		if err := r.deleteJob(ctx, scheduler); err != nil {
-			scheduler.Status.MarkJobNotReady("JobDeleteFailed", "Failed to delete CloudSchedulerSource job: %s", err.Error())
-			return err
-		}
-		scheduler.Status.MarkJobNotReady("JobDeleted", "Successfully deleted CloudSchedulerSource job: %s", scheduler.Status.JobName)
-
-		if err := r.PubSubBase.DeletePubSub(ctx, scheduler); err != nil {
-			return err
-		}
-
-		// Only set the jobName to empty after we successfully deleted the PubSub resources.
-		// Otherwise, we may leak them.
-		scheduler.Status.JobName = ""
-		removeFinalizer(scheduler)
-		return nil
-	}
-
-	// Ensure that there's finalizer there, since we're about to attempt to
-	// change external state with the topic, so we need to clean it up.
-	addFinalizer(scheduler)
+	scheduler.Status.ObservedGeneration = scheduler.Generation
 
 	topic := resources.GenerateTopicName(scheduler)
 	_, _, err := r.PubSubBase.ReconcilePubSub(ctx, scheduler, topic, resourceGroup)
 	if err != nil {
-		return err
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledPubSubFailedReason, "Reconcile PubSub failed with: %s", err)
 	}
 
 	jobName := resources.GenerateJobName(scheduler)
 	err = r.reconcileJob(ctx, scheduler, topic, jobName)
 	if err != nil {
-		scheduler.Status.MarkJobNotReady("JobReconcileFailed", "Failed to reconcile CloudSchedulerSource job: %s", err.Error())
-		return err
+		scheduler.Status.MarkJobNotReady(reconciledFailedReason, "Failed to reconcile CloudSchedulerSource job: %s", err.Error())
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledFailedReason, "Reconcile Job failed with: %s", err)
```

Comment:
err.Error()

Reply:
here and other places where we are passing just err with %s format...

Reply:
done